internal/mailer/validate.go

```diff
@@ -267,6 +279,20 @@ func (ev *EmailValidator) validateHost(ctx context.Context, host string) error {
 	return ErrInvalidEmailDNS
 }
 
+// Justification for this function is based on automated signups at scale.
+// The goal is to reduce the number of automated signups from bots
+func (ev *EmailValidator) validateMX(ctx context.Context, host string) error {
```

Comment:
This check should be in the validateHost code so we don't run two separate mx lookups for the same host. 

You may modify the validateHost to check the lookup there. Be sure to check both the LookupMX and LookupHost. (Check comments about RFC).